src/btree/bt_curnext.c

```diff
@@ -904,8 +904,16 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
         if (page != NULL &&
           (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD ||
             (newpage && cbt->page_deleted_count > 0))) {
-            WT_ERR(__wt_page_dirty_and_evict_soon(session, cbt->ref));
-            WT_STAT_CONN_INCR(session, cache_eviction_force_delete);
+            /* If checkpoint is happening on the btree, we can only evict clean content. */
+            if (__wt_btree_syncing_by_other_session(session)) {
+                if (!__wt_page_is_modified(page)) {
+                    __wt_page_evict_soon(session, cbt->ref);
+                    WT_STAT_CONN_INCR(session, cache_eviction_force_delete);
```

Comment:
Are we supposed to increment this stat when not force evicting something?

Reply:
Isn't it the original behaviour to have this stat incremented when we (dirty the page and) mark it to be evicted soon?

Reply:
Does marking something to be evicted soon equal forced eviction?

Reply:
I think the forced eviction happens elsewhere and this simply makes it a likely next candidate. This means that the stat name is kinda right and kinda wrong. Anyway not the problem for this PR.